nss/aad_auth/getent.go

```diff
@@ -0,0 +1,144 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/ubuntu/aad-auth/internal/cache"
+	"github.com/ubuntu/aad-auth/internal/logger"
+	"github.com/ubuntu/aad-auth/internal/nss"
+	"github.com/ubuntu/aad-auth/internal/nss/group"
+	"github.com/ubuntu/aad-auth/internal/nss/passwd"
+	"github.com/ubuntu/aad-auth/internal/nss/shadow"
+)
+
+// GetEnt processes the args and queries the database for the requested entries.
+func GetEnt(ctx context.Context, dbName, key string, cacheOpts ...cache.Option) (entries []string, err error) {
```

Comment:
I think we should rename the function again then. GetEnt should return the queried entities rather than a formatted string with the output, shouldn't it? Doesn't make much sense calling GetEnt and receiving a string with the error code and etc.

Reply:
Yes, let’s open a dictionary and pick a random name? :)

Also, you can argue the other way: GetEnt is what is the argument passed to the command line, so it’s the name, but maybe with the `Getent` case?

Reply:
let’s use that name :)